pywavefront/obj.py

```diff
@@ -336,6 +352,9 @@ def consume_faces(self):
             # Do we need to triangulate? Each line may contain a varying amount of elements
             triangulate = (len(self.values) - 1) > 3
 
+            if consumed_faces is not None:
+                consumed_faces.append([None] * (len(self.values) - 1))
```

Comment:
Personally I would collect the face list as triangles only. Then we know each face is 3 elements in the face list. Right now for some models it would be impossible to understand the face list as some ``f`` statements can contain a variable amount of indices for each ``f`` statement (3 or more). Ensure you ``append`` inside the triangulate check as well.

I would also keep it simple and just ``append()`` to ``consumed_faces`` instead of pre-allocate space like this. Python lists over-allocate internally so memory allocations will be less and less frequent as the list increases in size. It makes more sense if you pre-allocate a lot of indices with ``list.extend``.

Reply:
> Right now for some models it would be impossible to understand the face list as some f statements can contain a variable amount of indices for each f statement (3 or more).

Yes, this was actually intended. I am not sure how much preprocessing this library is supposed to do. Maybe someone (like me) just wants to access the raw face data from a Wavefront file. Especially, I could not find any other Python Wavefront parsing library.
In #73 it sounded a lot that this library is mainly targeted at "preparing interleaved vertex data". I described my use case [in this comment at #73](https://github.com/greenmoss/PyWavefront/issues/73#issue-346060142). I might be lucky that my Wavefront file only has triangle faces anyway, but if it had quad faces, I would have rolled my own library and thus duplicating a lot of raw Wavefront parsing. This seems unnecessary to me this project could distinguish between "raw parsing" and "preprocessing".

> I would also keep it simple and just append() to consumed_faces instead of pre-allocate space like this.
Note that I don't preallocate the overall face list, but I preallocate the list of vertex indices of each and every face, i.e. I preallocate lists of 3 elements most of the time for my Wavefront files.  So no amortization effects apply. But sure, one might call this "premature optimization". I'd argue in favor of keeping it because it is a very trivial line and also not decreasing readability/maintainbility.